test/functional/blockchain.py

```diff
@@ -44,6 +45,38 @@ def run_test(self):
         self._test_stopatheight()
         assert self.nodes[0].verifychain(4, 0)
 
+    def _test_getblockchaininfo(self):
+        keys = [
+            'bestblockhash',
+            'bip9_softforks',
+            'blocks',
+            'chain',
+            'chainwork',
+            'difficulty',
+            'headers',
+            'mediantime',
+            'pruned',
+            'softforks',
+            'verificationprogress',
+        ]
+        res = self.nodes[0].getblockchaininfo()
+        # should have exact fields
+        assert_equal(sorted(res.keys()), keys)
+
```

Comment:
Should assert values or types of each field?

Reply:
value: only if you're testing a meaningful value
type: I don't think we do this for other RPC methods, so I wouldn't bother

Thinking about this some more, perhaps we can add some utility method `check_rpc_return_object(dict)` which tests the presence and type of each field in the return object? Obviously not for this PR.